src/components/ContactForm/index.js

```diff
@@ -0,0 +1,149 @@
+    setMessage(event.target.value);
+  };
+
+  const renderInput = (title, id, error, handleChange) => {
+    const styles = classNames("input", {
+      border: error,
+    });
+
+    return (
+      <>
+        <Typography weight="medium" variant="small-body" color="oxford-blue">
+          {title}
+        </Typography>
+        <input
+          onChange={handleChange}
+          type="text"
+          name={id}
+          id={id}
+          data-name={title}
+          styleName={styles}
+        />
+      </>
+    );
+  };
+
+  const renderName = () => {
+    const error = !name && enableErrors;
+    const errorStyles = classNames("error", {
+      visible: error,
+    });
+
+    return (
+      <label htmlFor="name" styleName="label">
+        {renderInput("Name", "name", error, onChangeName)}
+        <div styleName={errorStyles}>
+          <Typography variant="xsmall-body" color="klein-blue">
+            Please enter your name
+          </Typography>
+        </div>
+      </label>
+    );
+  };
+
+  const renderEmail = () => {
+    const error = !validEmail(email) && enableErrors;
+    const errorStyles = classNames("error", {
+      visible: error,
+    });
+
+    return (
+      <label htmlFor="email" styleName="label">
+        {renderInput("Email", "email", error, onChangeEmail)}
+        <div styleName={errorStyles}>
+          <Typography variant="xsmall-body" color="klein-blue">
+            Please enter a valid email address
+          </Typography>
+        </div>
+      </label>
+    );
+  };
+
+  const renderMessage = () => {
+    const error = !message && enableErrors;
+    const styles = classNames("textarea", {
+      border: error,
+    });
+    const errorStyles = classNames("error", {
+      visible: error,
+    });
+
+    return (
+      <label htmlFor="message" styleName="message">
+        <Typography weight="medium" variant="small-body" color="oxford-blue">
+          Message
+        </Typography>
+        <textarea
+          onChange={onChangeMessage}
+          name="message"
+          id="message"
+          rows="5"
+          data-name="Message"
+          styleName={styles}
+        />
+        <div styleName={errorStyles}>
+          <Typography variant="xsmall-body" color="klein-blue">
+            Please write your message
+          </Typography>
+        </div>
+      </label>
+    );
+  };
+
+  const handleClick = () => {
+    setEnableErrors(true);
+
+    setEnableSubmit(validEmail(email) && message && name);
+  };
+
+  return (
+    <form
+      id="contact-form"
+      name="contact-form"
+      data-name="Contact Form"
+      method="post"
+      netlify-honeypot="bot-field"
+      data-netlify="true"
+    >
+      <input type="hidden" name="bot-field" />
+      {renderName()}
+      {renderEmail()}
+      {renderMessage()}
+      <div styleName="button">
+        <Button
+          handleClick={handleClick}
+          type={enableSubmit ? "submit" : "button"}
```

Comment:
why do we need to change the button type?

Reply:
To prevent the submission of the form before all values are filled and well-formatted.

Reply:
Can't we use a `disable` property? I've never seen this approach to prevent submission

Reply:
Not really, I discuss the disabled approach with Inês. 
As we don't have a proper way to know when the user finishes writing to display errors, we decide to let the user click the button to check and display errors. And we only submit the form when the values are filled and email well-formated. 
Do you have another suggestion?

Reply:
I've just found a better way to deal with errors, using form `onSubmit`. I'll try

Reply:
Not entirely sure but maybe having a handleSubmit function with `event.preventDefault();` when the fields are not correctly filled could work.

Reply:
Yes, it's exactly that. I've just updated it. 
Thanks

Reply:
Glad to help